components/ArticleCategories/AddCategoryDialog.js

```diff
@@ -155,26 +87,23 @@ function AddCategoryDialog({
       maxWidth="lg"
     >
       <DialogTitle id="add-category-dialog-title">{t`Categorize this message`}</DialogTitle>
-      <DialogContent className={classes.dialogContent}>
-        <Grid container spacing={1}>
+      <DialogContent>
+        <Hint>
+          {t`Articles are mostly categorized by AI based on our current data, but you can provide your own opinion to improved the categorization.`}
+        </Hint>
+        <div>
           {allCategories.map(category => (
-            <Grid
-              item
+            <CategoryOption
               key={category.id}
-              xs={12}
-              md={6}
-              lg={4}
-              className={classes.gridItem}
-            >
-              <CategoryOption
-                category={category}
-                onAdd={handleAdd}
-                selected={isSelectedMap[category.id]}
-                disabled={loading}
-              />
-            </Grid>
+              articleId={articleId}
```

Comment:
Since addition, deletion, upvote and downvote all happens within the option, I think it is more logical if we handle these logic together -- we should handle all these actions either within `<CategoryOption>`, or handling them all in `<AddCategoryDialog>` and put these callback prop on `<CategoryOption>`.

As we pass `articleId` down to `CategoryOption` to handle its deletion, upvote and downvote, I think its good to move addition logic down to `CategoryOption` as well.